packages/grid/x-data-grid/src/hooks/features/filter/gridFilterState.ts

```diff
@@ -21,13 +21,13 @@ export interface GridFilterState {
    * If a row is not registered in this lookup, it is filtered.
    * This is the equivalent of the `visibleRowsLookup` if all the groups were expanded.
    */
-  filteredRowsLookup: Record<GridRowId, boolean>;
+  filteredRowsLookup: Set<GridRowId>;
   /**
    * Visibility status for each row.
    * A row is visible if it is passing the filters AND if its parents are expanded.
    * If a row is not registered in this lookup, it is visible.
    */
-  visibleRowsLookup: Record<GridRowId, boolean>;
+  visibleRowsLookup: Set<GridRowId>;
```

Comment:
TODO: Create new selectors for these that select the `Set`, and update the old selectors to return an `object` like they do now, otherwise it's a breaking change.